File: src/eQuantic.UI.Runtime/src/shared/components/BottomNavigation.ts
import { $eq, Badge, Box, BoxStyle, BuildContext, ColorToken, Column, Component, ComponentContext, CornerRadii, Flexible, HtmlElement, Icon, IconSize, NavItem, Positioned, Pressable, Radius, Row, SizeValue, Stack, StatelessComponent, Text, TypeStyle, VariantColors, VisualNode } from "../runtime-exports";

export class BottomNavigation extends StatelessComponent {
    constructor(items?: any, selected?: any, onSelect: any = null, props?: any) {
        super(props);
        if (items !== undefined) this.items = items;
        if (selected !== undefined) this.selected = selected;
        if (onSelect !== undefined) this.onSelect = onSelect;
        this.items = items;this.selected = selected;this.onSelect = onSelect;
    }

    build(context: BuildContext) {
        let theme = context.theme;let primary = theme.colors('primary');if ((this.items.length < 3 || this.items.length > 5)) throw new Error('BottomNavigation takes 3-5 destinations (spec B4): 2 → Tabs, 6+ → Drawer.');let row = new Row(0, { width: SizeValue.fill, height: SizeValue.fill });for (let i = 0; i < this.items.length; i++) {let item = this.items[i];let isActive = i === this.selected;let index = i;let filled; let glyph = isActive && ((item.selectedIcon != null) && (filled = item.selectedIcon, true)) ? filled : item.icon;let tint = isActive ? primary.onSubtle : theme.textMuted;let iconNode = new Icon(glyph, IconSize.md, tint);if (item.badgeCount > 0) {let stack = new Stack();stack.add(iconNode);stack.add(new Positioned(new Badge(item.badgeCount, 99, 'destructive', { ring: true }), -4, -8));iconNode = stack;}let pillContent = new Row(0, { main: 'center', height: SizeValue.fill });pillContent.add(iconNode);let pill = new Box(new BoxStyle({ width: 56, height: 26, background: isActive ? primary.subtle : null, cornerRadius: new CornerRadii(Radius.full) }), pillContent);let column = new Column(2, { height: SizeValue.fill, main: 'center', cross: 'center' });column.add(pill);column.add(new Text(item.label, 'caption', tint, 1, { styleOverride: new TypeStyle(11, 14, 'bold', 0.1, 1.3) }));row.add(new Flexible(new Pressable(column, this.onSelect === null ? null : () => this.onSelect(index), { label: item.label, pressedBackground: theme.surfaceSubtle })));}return new Box(new BoxStyle({ width: SizeValue.fill, height: 56, background: theme.surface }), row);
    }

}

File: src/eQuantic.UI.Components.Shared/BottomNavigation.cs
using eQuantic.UI.Primitives;

namespace eQuantic.UI.Components.Shared;

/// <summary>One BottomNavigation destination (spec B4): glyph pair (outline/filled) + a REQUIRED
/// label; <see cref="BadgeCount"/> anchors a count Badge (B7) to the icon's top-end corner.</summary>
public sealed record NavItem(Icons Icon, string Label, Icons? SelectedIcon = null)
{
    public int BadgeCount { get; init; }
}

/// <summary>
/// The design system's BottomNavigation (spec B4), CONTROLLED: 3–5 destinations (2 → use Tabs;
/// 6+ → Drawer; out of range throws), equal-width columns where the WHOLE column is the target.
/// Item = icon in a 56×26 pill slot + 11/700 label — labels are ALWAYS visible (icon-only nav
/// fails review). Active: Primary-subtle pill + filled glyph + OnSubtle label; inactive: TextMuted
/// outline glyph. Badges anchor to the icon's top-end. v1 fences: the E2 top shadow joins the
/// engine shadow primitive; the bottom safe-area inset joins the host insets.
/// </summary>
public sealed class BottomNavigation : StatelessComponent
{
    public BottomNavigation(IReadOnlyList<NavItem> items, int selected, Action<int>? onSelect = null)
    {
        Items = items;
        Selected = selected;
        OnSelect = onSelect;
    }

    public IReadOnlyList<NavItem> Items { get; init; }
    public int Selected { get; init; }
    public Action<int>? OnSelect { get; init; }

    public override VisualNode Build(ComponentContext context)
    {
        var theme = context.Theme;
        var primary = theme.Colors(Variant.Primary);
        if (Items.Count is < 3 or > 5)
            throw new ArgumentException("BottomNavigation takes 3-5 destinations (spec B4): 2 → Tabs, 6+ → Drawer.");

        var row = new Row(gap: 0) { Width = SizeValue.Fill, Height = SizeValue.Fill };
        for (var i = 0; i < Items.Count; i++)
        {
            var item = Items[i];
            var isActive = i == Selected;
            var index = i;

            var glyph = isActive && item.SelectedIcon is { } filled ? filled : item.Icon;
            var tint = isActive ? primary.OnSubtle : theme.TextMuted;

            VisualNode iconNode = new Icon(glyph, IconSize.Md, tint);
            if (item.BadgeCount > 0)
            {
                var stack = new Stack();
                stack.Add(iconNode);
                stack.Add(new Positioned(new Badge(item.BadgeCount) { Ring = true }, top: -4, end: -8));
                iconNode = stack;
            }

            var pillContent = new Row(gap: 0) { Main = MainAlign.Center, Height = SizeValue.Fill };
            pillContent.Add(iconNode);
            var pill = new Box(new BoxStyle
            {
                Width = 56,
                Height = 26,
                Background = isActive ? primary.Subtle : null,
                CornerRadius = new CornerRadii(Radius.Full),
            }, pillContent);

            var column = new Column(gap: 2)
            {
                Height = SizeValue.Fill,
                Main = MainAlign.Center,
                Cross = CrossAlign.Center,
            };
            column.Add(pill);
            column.Add(new Text(item.Label, TypeRole.Caption, tint, maxLines: 1)
            {
                StyleOverride = new TypeStyle(11, 14, FontWeight.Bold, 0.1f, 1.3f),
            });

            row.Add(new Flexible(new Pressable(column, OnSelect is null ? null : () => OnSelect(index))
            {
                Label = item.Label,
                PressedBackground = theme.SurfaceSubtle,
            }));
        }

        return new Box(new BoxStyle
        {
            Width = SizeValue.Fill,
            Height = 56,
            Background = theme.Surface,
        }, row);
    }
}
